fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/impl/transfer/item/CursorSlotWrapper.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.transfer.item;
+
+import java.util.Map;
+
+import com.google.common.collect.MapMaker;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.screen.ScreenHandler;
+
+import net.fabricmc.fabric.api.transfer.v1.item.ItemVariant;
+import net.fabricmc.fabric.api.transfer.v1.storage.StoragePreconditions;
+import net.fabricmc.fabric.api.transfer.v1.storage.base.SingleSlotStorage;
+import net.fabricmc.fabric.api.transfer.v1.transaction.TransactionContext;
+import net.fabricmc.fabric.api.transfer.v1.transaction.base.SnapshotParticipant;
+
+public class CursorSlotWrapper extends SnapshotParticipant<ItemStack> implements SingleSlotStorage<ItemVariant> {
+	private static final Map<ScreenHandler, CursorSlotWrapper> WRAPPERS = new MapMaker().weakKeys().makeMap();
+
+	public static CursorSlotWrapper get(ScreenHandler screenHandler) {
+		return WRAPPERS.computeIfAbsent(screenHandler, CursorSlotWrapper::new);
+	}
+
+	private final ScreenHandler screenHandler;
+
+	private CursorSlotWrapper(ScreenHandler screenHandler) {
```

Comment:
might wana pass the player as well, iirc in 1.16 the cursor slot was on the player rather than the screen handler. They may change it back, or it'll be important if we backport the api

Reply:
Will just be regular MC update breakage imo.
Re backporting: it's useless imo cause no mods will support it.

Reply:
Fyi in 1.16 it didn't require screen handlers... these kinds of changes are impossible to predict, and we should just have as few parameters as possible.